packages/core/test/client.test.ts

```diff
@@ -72,6 +72,44 @@ describe('@bugsnag/core/client', () => {
       expect(client).toBeTruthy()
       expect(logger.warn.mock.calls.length).toBe(0)
     })
+
+    it('prevents sendSession() with invalid endpoint config', done => {
+      // @ts-expect-error
+      const client = new Client({ apiKey: 'API_KEY', endpoints: { sessions: '/sessions' } })
+      const sendSession = jest.fn()
+
+      client._setDelivery(client => ({
+        sendSession: () => {
+          sendSession()
+          done('sendSession() should not be called')
+        },
+        sendEvent: () => { }
+      }))
+
+      client.startSession()
+
+      process.nextTick(() => {
+        expect(sendSession).not.toHaveBeenCalled()
+        done()
+      })
+    })
+
+    it('prevents sendEvent() with invalid endpoint config', done => {
+      // @ts-expect-error
+      const client = new Client({ apiKey: 'API_KEY', endpoints: { sessions: '/sessions' } })
+
+      client._setDelivery(client => ({
+        sendSession: () => { },
+        sendEvent: () => { done('sendEvent() should not be called') }
+      }))
+
+      client.notify(new Error('111'), undefined, (err, event) => {
+        expect(err).toBe(null)
+        expect(event).toBeTruthy()
+        expect(event.errors[0].errorMessage).toBe('111')
+        done()
```

Comment:
could we have a `expect(sendEvent).not.toHaveBeenCalled()` here?

Reply:
sure - I think it's redundant with `sendEvent: () => { done('sendEvent() should not be called') }` but it won't hurt.